android/src/main/java/com/swmansion/reanimated/nodes/FormatNode.java

```diff
@@ -0,0 +1,31 @@
+package com.swmansion.reanimated.nodes;
+
+import com.facebook.react.bridge.ReadableMap;
+import com.swmansion.reanimated.NodesManager;
+import com.swmansion.reanimated.Utils;
+import java.text.NumberFormat;
+import java.text.DecimalFormat;
+
+public class FormatNode extends Node {
+  private NumberFormat nf;
+  private final int mInputID;
+
+  public FormatNode(int nodeID, ReadableMap config, NodesManager nodesManager) {
+    super(nodeID, config, nodesManager);
+    mInputID = config.getInt("value");
+    nf = new DecimalFormat(config.getString("format"));
+  }
+
+  @Override
+  protected String evaluate() {
+    StringBuilder builder = new StringBuilder();
+    Node inputNode = mNodesManager.findNodeById(mInputID, Node.class);
+    Object value = inputNode.value();
+    if (value instanceof Double) {
+      builder.append(nf.format(value));
```

Comment:
return here or assign `nf.format(val)` to some `result` variable (same commend as in the iOS code). There is no point in allocating string builder and appending only once to an empty string